lib/tty/prompt/question.rb

```diff
@@ -35,23 +35,33 @@ def to_s
       #
       # @api public
       def initialize(prompt, **options)
-        @prompt     = prompt
-        @prefix     = options.fetch(:prefix) { @prompt.prefix }
-        @default    = options.fetch(:default) { UndefinedSetting }
-        @required   = options.fetch(:required) { false }
-        @echo       = options.fetch(:echo) { true }
-        @in         = options.fetch(:in) { UndefinedSetting }
-        @modifier   = options.fetch(:modifier) { [] }
-        @validation = options.fetch(:validate) { UndefinedSetting }
-        @convert    = options.fetch(:convert) { UndefinedSetting }
+        # Option deprecation
+        if options.fetch(:validation) { nil }
+          warn <<-MSG.gsub(/[[:space:]]+/, " ").strip
+            [DEPRECATION] The `:validation` option is deprecated and
+            will be removed in a future release of tty-prompt
+            (use functionally identical `:validate` instead)
+          MSG
```

Comment:
I do like this message but wonder if we could make it a bit shorter. How about?
```text
[DEPRECATION] The `:validation` option is deprecated. Use `:validate` instead.
```
I feel as though we don't need to mention that it will be removed in the future.